app/controllers/miq_ae_class_controller.rb

```diff
@@ -1628,6 +1713,45 @@ def refresh_git_domain
 
   private
 
+  def playbook_inputs(method)
+    existing_inputs = method.inputs
+    new_inputs = params[:extra_vars]
+    inputs_to_save = []
+    inputs_to_delete = []
+    new_inputs.each do |i, input|
+      field = input.length == 4 ? MiqAeField.find_by(:id => input.last) : MiqAeField.new
+      field.name = input[0]
+      field.default_value = input[1] == "" ? nil : input[1]
+      field.datatype = input[2]
+      field.priority = i
+      inputs_to_save.push(field)
+    end
+    existing_inputs.each do |existing_input|
+      inputs_to_delete.push(existing_input.id) unless inputs_to_save.any? { |i| i.id == existing_input.id }
+    end
+    return inputs_to_save, inputs_to_delete
+  end
+
+  def set_playbook_data
+    data = {
+      :repository_id  => params['repository_id'],
+      :playbook_id    => params['playbook_id'],
+      :credential_id  => params['credential_id'],
+      :become_enabled => params['become_enabled'],
+      :verbosity      => params['verbosity'],
+    }
+    data[:network_credential_id] = params['network_credential_id'] if params['network_credential_id']
+    data[:cloud_credential_id] = params['cloud_credential_id'] if params['cloud_credential_id']
```

Comment:
`copy_params_if_set(data, params, %i(network_credential_id cloud_credential_id))` actually you can do it will all the params, right?
`data = copy_params_if_set({}, params, %i(repository_id ....... network_credential_id cloud_credential_id))`